code/modules/mob/living/carbon/xenomorph/xenomorph.dm

```diff
@@ -259,4 +259,31 @@
 	return
 
 /mob/living/carbon/Xenomorph/reagent_check(datum/reagent/R) //For the time being they can't metabolize chemicals.
-	return TRUE
\ No newline at end of file
+	return TRUE
+
+/mob/living/carbon/Xenomorph/update_leader_tracking(mob/living/carbon/Xenomorph/X)
+	if(!hud_used || !hud_used.locate_leader)
+		return
+
+	var/obj/screen/LL_dir = hud_used.locate_leader
+	if(!X || xeno_caste.caste_flags & CASTE_IS_INTELLIGENT)
```

Comment:
Is there reasonable expectation of this to be called without an `X`?

Reply:
Should never be called outside the subsystem really, so no. Will remove.